core/src/main/java/it/pagopa/selfcare/dashboard/core/UserV2ServiceImpl.java

```diff
@@ -133,10 +134,21 @@ public String createUsers(String institutionId, String productId, UserToCreate u
     public void addUserProductRoles(String institutionId, String productId, String userId, Set<String> productRoles) {
         log.trace("createOrUpdateUserByUserId start");
         log.debug("createOrUpdateUserByUserId userId = {}", userId);
+        verifyOnboardingStatus(institutionId, productId);
         List<CreateUserDto.Role> role = retrieveRole(productId, productRoles);
         userApiConnector.createOrUpdateUserByUserId(institutionId, productId, userId, role);
         log.trace("createOrUpdateUserByUserId end");
     }
+
+    private void verifyOnboardingStatus(String institutionId, String productId) {
+        Institution institution = msCoreConnector.getInstitution(institutionId);
+        if (institution.getOnboarding() == null || institution.getOnboarding().stream()
+                .noneMatch(onboarding -> onboarding.getProductId().equals(productId) && onboarding.getStatus().equals(RelationshipState.ACTIVE))
```

Comment:
might a comment here help clarify this condition?

Reply:
added